File: ios/Types/FetchInit.swift
import ExpoModulesCore

struct FetchInit: Record {
  @Field var body: String?
  @Field var headers: [String: String]?
  @Field var method: String?
  @Field var redirect: String?
  @Field var timeout: Double?
  @Field var exposeNonHttpOnlyCookies: Bool?
}

File: android/src/main/java/expo/modules/fetcher/types/FetchInit.kt
package expo.modules.fetcher.types

import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

class FetchInit : Record {
  @Field
  val body: Any? = null

  @Field
  val headers: Map<String, String>? = null

  @Field
  val method: String? = null

  @Field
  val redirect: String? = null

  @Field
  val timeout: Double? = null
}
